src/internal/clusterstate/v2.6.0/clusterstate.go

```diff
@@ -50,6 +51,24 @@ func Migrate(state migrations.State) migrations.State {
 			return pfsdb.SetupCommitProvenanceV0(ctx, env.Tx)
 		}).
 		Apply("Remove Alias Commits", func(ctx context.Context, env migrations.Env) error {
+			// locking the following tables is necessary for the following 2 migration "Apply"s:
+			// "Remove Alias Commits", "Remove branch from the Commit key"
+			for _, table := range []string{
+				"collections.repos",
+				"collections.branches",
+				"collections.commits",
+				"pfs.commit_diffs",
+				"pfs.commit_totals",
+				"storage.tracker_objects",
+				"pfs.commits",
+				"pfs.commit_provenance",
+				"collections.pipelines",
+				"collections.jobs",
+			} {
+				if _, err := env.Tx.ExecContext(ctx, fmt.Sprintf("LOCK TABLE %s IN EXCLUSIVE MODE", table)); err != nil {
```

Comment:
We should probably add a migrations.Env.LockTable() method, so that we can skip locks during `test-migrations`s.